src/iguana/algorithms/clas12/FiducialFilter/Algorithm.cc

```diff
@@ -0,0 +1,294 @@
+    }    
+    
+  bool FiducialFilter::DC_fiducial_cut_theta_phi_pass1(FiducialFilter::traj_row_data const traj_row, int const torus, int const pid) const{
+      
+      const auto minparams = ((torus==-1) ? minparams_in_theta_phi_pass1 : minparams_out_theta_phi_pass1);
+      const auto maxparams = ((torus==-1) ? maxparams_in_theta_phi_pass1 : maxparams_out_theta_phi_pass1);
+      double theta_DCr = 5000;
+      double phi_DCr_raw = 5000;
+      double x=0;
+      double y=0;
+      double z=0;
+      for(int r = 0; r<3; r++){  
+        x=0;y=0;z=0;
+        switch(r){
+        case 0:
+          x=traj_row.x1;
+          y=traj_row.y1;
+          z=traj_row.z1;
+          break;
+        case 1:
+          x=traj_row.x2;
+          y=traj_row.y2;
+          z=traj_row.z2;
+          break;
+        case 2:
+          x=traj_row.x3;
+          y=traj_row.y3;
+          z=traj_row.z3;
+          break;
+        }
+        int sector = traj_row.sector;
+
+        theta_DCr = 180 / PI * acos(z / sqrt(pow(x,2) + pow(y,2) + pow(z,2)));
+        phi_DCr_raw = 180 / PI * atan2(y / sqrt(pow(x,2) + pow(y,2) + pow(z,2)), 
+                       x /sqrt(pow(x,2) + pow(y,2) + pow(z,2)));
+
+        double phi_DCr = 5000;
+
+        if (sector == 1) phi_DCr = phi_DCr_raw;
+        if (sector == 2) phi_DCr = phi_DCr_raw - 60;
+        if (sector == 3) phi_DCr = phi_DCr_raw - 120;
+        if (sector == 4 && phi_DCr_raw > 0) phi_DCr = phi_DCr_raw - 180;
+        if (sector == 4 && phi_DCr_raw < 0) phi_DCr = phi_DCr_raw + 180;
+        if (sector == 5) phi_DCr = phi_DCr_raw + 120;
+        if (sector == 6) phi_DCr = phi_DCr_raw + 60;
+
+        int this_pid = 0;
+
+        switch (pid)
+          {
+          case 11: this_pid = 0; break;
+          case 2212: this_pid = 1; break;
+          case 211: this_pid = 2; break;
+          case -211: this_pid = 3; break;
+          case 321: this_pid = 4; break;
+          case -321: this_pid = 5; break;
+          default: return false; break;
+          }
+
+
+
+        double calc_phi_min = minparams[this_pid][sector - 1][r][0] + minparams[this_pid][sector - 1][r][1] * std::log(theta_DCr) 
+          + minparams[this_pid][sector - 1][r][2] * theta_DCr + minparams[this_pid][sector - 1][r][3] * theta_DCr * theta_DCr;
+
+        double calc_phi_max = maxparams[this_pid][sector - 1][r][0] + maxparams[this_pid][sector - 1][r][1] * std::log(theta_DCr)
+          + maxparams[this_pid][sector - 1][r][2] * theta_DCr + maxparams[this_pid][sector - 1][r][3] * theta_DCr * theta_DCr;
+
+        if(std::isnan(calc_phi_min)||std::isnan(calc_phi_max)) return false;
+        if((phi_DCr < calc_phi_min) || (phi_DCr > calc_phi_max)) return false;
+      }
+      return true;
+  }
+  std::map<int, FiducialFilter::traj_row_data> FiducialFilter::GetTrajMap(hipo::bank const &bank) const
+  {
+      std::map<int, FiducialFilter::traj_row_data> traj_map;
+      
+      for(int row = 0; row < bank.getRows(); row++){
```

Comment:
Do you care about upstream `REC::Traj` filters (`getRowList()`)? We don't have any yet, but we could someday...

Reply:
What would the scenario look like where users would be filtering out REC::Traj rows without subsequently filtering out the corresponding REC::Particle? To be safe, would you suggest we just call the filtered REC::Traj bank? I can add this with a quick commit.

Reply:
Filters should always be assumed to be used. In this case, since this algorithm filters only `REC::Particle`, it's just a matter of performance; but if you don't use the filtered version, it could someday be one of those "subtle bugs"...

Reply:
I agree. Looping is now over the `getRowList()` as of e064db64b3350ca38ddc071318e5a91de22ce318